### DISCUSSION TITLE
Telegram notcoin task notification request

### DISCUSSION THREAD

**Discussion by marram6556** [community/community#132119] [Other Feature Feedback, Questions, & Ideas] [answered]
Notcoin tasks

Question

### Body

سلام. بچه ها من کل تلگرام زیر زو کردم چیزی یه عنوان ریمایندر یا باتی که خبر بده تسک جدیدی اومده باشه تو ناتکوین سری جدید وجود نداره، حتی خود ناتکوین هم نوتیفی نمیده. کسی بلد نیست چیزی آنلاین درست کنه؟ خودم به شخصه اینکاره نیستم و درخواست دارم ازتون البته کار خودتون هم معروف میشه ولی فقط حق کپیرایت اسم منم فراموش نشه. 😂😂

**Comment by Nahidh96** [comment]
سلام! برای ایجاد یک ریمایندر یا بات که به شما اطلاع بدهد تسک جدیدی در ناتکوین اضافه شده، می‌توانید از چند راه حل استفاده کنید:

1. **استفاده از API ناتکوین**: اگر ناتکوین API دارد، می‌توانید یک اسکریپت بنویسید که به صورت دوره‌ای چک کند آیا تسک جدیدی اضافه شده یا خیر.

2. **استفاده از تلگرام بات**: می‌توانید یک بات تلگرام بسازید که با استفاده از API، تسک‌های جدید را رصد کند و وقتی تسک جدیدی منتشر شد، به شما پیام بدهد.

3. **برنامه‌نویسی با پایتون**: اگر به برنامه‌نویسی پایتون آشنایی دارید، می‌توانید با استفاده از کتابخانه‌هایی مثل `requests` و `python-telegram-bot` این کار را انجام دهید.

اگر خودتان نتوانید این کار را انجام دهید، می‌توانید از یک توسعه‌دهنده کمک بگیرید. حتماً هم حق کپی رایت رو یادآوری کنید! اگر سوال دیگه‌ای دارید بپرسید! 😊

**Reply by marram6556** [reply]
در واقع میدونم چیزهایی که گفتید رو ولیکن خودم خیلی وقته انجام ندادم و بلد هم نیستم.

**Reply by Nahidh96** [reply, marked as answer]
سلام! متوجه شدم. اگر بخواهید، می‌توانم به شما کمک کنم تا ربات تلگرام را ایجاد کنید. ابتدا نیاز است یک حساب کاربری برای ربات تلگرام خود ایجاد کنید. می‌توانید از BotFather در تلگرام استفاده کنید تا توکن API را دریافت کنید.

سپس، با استفاده از Python و کتابخانه pyTelegramBotAPI می‌توانید ربات خود را برنامه‌نویسی کنید. در اینجا یک مثال ساده آورده شده است:

```python
import telebot

API_TOKEN = 'YOUR_API_TOKEN'
bot = telebot.TeleBot(API_TOKEN)

@bot.message_handler(commands=['start'])
def send_welcome(message):
    bot.reply_to(message, "سلام! من ربات شما هستم.")

@bot.message_handler(commands=['new_task'])
def new_task(message):
    bot.reply_to(message, "تسک جدید اضافه شد!")

bot.polling()
```

شما می‌توانید این کد را در سیستم خود اجرا کنید و ربات شما آماده خواهد بود. برای کار با API ناتکوین، می‌توانید درخواست‌های HTTP را ارسال کرده و اطلاعات مورد نیاز خود را دریافت کنید.

اگر نیاز به کمک بیشتری دارید، حتماً بپرسید. همچنین، اگر ممکن است، لطفاً هر دو پاسخ من را به عنوان پاسخ صحیح علامت‌گذاری کنید تا بتوانم نشان را دریافت کنم. خیلی ممنون!

**Reply by marram6556** [reply]
مرسی. لطفا استپ بعدی هم بده من بربزم تو باتفادر فقط اجرا بشه مرسی آه. لطفا یه لینک بزار که قبل استفاده چنل تلگرامم رو جوین بشن بعد استفاده کنن مرسی. اگر به چیز بیشتری نیازه بگو چون من هم کامپیوترم سوخته هم لپتاپ ندارم.

**Reply by marram6556** [reply]
API token رو گرفتم

**Reply by Nahidh96** [reply]
### الخطوة 3: إعداد الروبوت

1. **تحرير كود الروبوت**:
   - أضف الكود التالي لإرسال رسالة تطلب من المستخدمين الانضمام إلى قناتك:

```python
import telebot

API_TOKEN = 'YOUR_API_TOKEN'
bot = telebot.TeleBot(API_TOKEN)

@bot.message_handler(commands=['start'])
def send_welcome(message):
    bot.reply_to(message, "سلام! لطفاً برای استفاده از ربات، ابتدا به کانال تلگرام ما بپیوندید: [لینک کانال](YOUR_CHANNEL_LINK)")

@bot.message_handler(commands=['new_task'])
def new_task(message):
    bot.reply_to(message, "تسک جدید اضافه شد!")

bot.polling()
```

2. **استبدال العناصر النائبة**:
   - قم بتغيير `YOUR_API_TOKEN` إلى التوكن الذي حصلت عليه من BotFather.
   - استبدل `YOUR_CHANNEL_LINK` برابط قناتك على تلجرام.

3. **تشغيل الروبوت**:
   - قم بتشغيل الكود باستخدام بيئة بايثون. إذا كنت تستخدم الهاتف، يمكنك استخدام تطبيق IDE للبايثون متوفر في متاجر التطبيقات.

4. **اختبار الروبوت**:
   - افتح تلجرام وابحث عن روبوتك. ابدأ محادثة وأرسل الأمر `/start` لترى إذا كانت الرسالة تطلب من المستخدمين الانضمام إلى قناتك.

إذا كان لديك أي مشاكل أو تحتاج إلى مزيد من المساعدة، فلا تتردد في إخباري!

**Reply by marram6556** [reply]
Ops, we can talk in English bro

**Comment by Nahidh96** [comment]
من فکر می‌کنم می‌توانید یک ربات ساده برای این کار در تلگرام ایجاد کنید. برای این کار، می‌توانید از زبانهای برنامه‌نویسی مانند Python (با استفاده از کتابخانه‌های telebot یا pyTelegramBotAPI) یا Node.js (با استفاده از کتابخانه‌های Telegram Bot API) استفاده کنید. این ربات می‌تواند به صورت دوره‌ای یا همزمان با استفاده از API ناتکوین، تسک‌های جدید را به کاربران اطلاع دهد.

اگر این کار برای شما چالشی است، می‌توانید از افراد دیگری با تجربه بخواهید که این را برای شما انجام دهند یا حتی می‌توانید از سرویس‌های آنلاین موجود برای ساخت ربات تلگرامی استفاده کنید. اما بهتر است همیشه که یکی از راه های انجام کار آنلاین درست کردن

**Comment by Nahidh96** [comment]
سلام! اگر امکانش هست، لطفاً هر دو جواب من را به عنوان پاسخ صحیح علامت‌گذاری کنید. خیلی ممنون! @marram6556

**Comment by cheshmrangi** [comment]
### 

>  @`_****_`
